### PR TITLE
Update `rustls` and enable `ring` feature needed by `UreqClient`

### PR DESCRIPTION
**Pull request by glmdev**
TLDR: Update `rustls` to latest version and enable the `ring` feature on it to fix 2 build errors.

--

Hello :)

Thanks for the good work on this library. I'm a software engineer, but relatively new to Rust, so let me know if I'm missing something.

I'm trying to use this API in a hobby project. I need to use the `UreqClient` instead of the `ReqwestClient` because I'm running w/in Rocket, which causes the `ReqwestClient` to panic when it tries to create a blocking `tokio` runtime.

However, when I try to use the `UreqClient`, I get the following build error:

```
error[E0599]: no function or associated item named `builder` found for struct `ClientConfig` in the current scope
   --> /home/garrettmills/.cargo/git/checkouts/p5x-proxmox-api-979e6aed6768e1f7/7d75408/proxmox-api/src/clients/ureq.rs:200:44
    |
200 |         let config = rustls::ClientConfig::builder()
    |                                            ^^^^^^^ function or associated item not found in `ClientConfig`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `proxmox-api` (lib) due to 1 previous error
```

I traced this down to the `Cargo.toml` in the `proxmox-api` crate not enabling the `ring` feature for `rustls`. After enabling that feature (on my fork), I got a _new_ error:

```
error[E0308]: mismatched types
   --> /home/garrettmills/.cargo/git/checkouts/p5x-proxmox-api-979e6aed6768e1f7/5020441/proxmox-api/src/clients/ureq.rs:206:34
    |
206 |             .tls_config(Arc::new(config))
    |                         -------- ^^^^^^ expected `rustls::client::client_conn::ClientConfig`, found `ClientConfig`
    |                         |
    |                         arguments to this function are incorrect
    |
    = note: `ClientConfig` and `rustls::client::client_conn::ClientConfig` have similar names, but are actually distinct types
```

From my testing, updating `rustls` to the latest version (`0.23.16`) resolves this.

### CODE REVIEW

**Comment by datdenkikniet** [comment]
Thanks for spotting! Seems like `ureq` did a cross-version bump on a semver compatible version... Oh well. CI is no longer very happy for unrelated reasons, and since this is a client-side-only change, I will merge it either way way.

Thanks!

**Comment by datdenkikniet** [comment]
Welp, okay, I messed that up: it seems to still not compile. Figured it was a small change & it looked right, but I was wrong. Also no step that actually tries to compile `ureq-client` in the CI, so this didn't quite work out the way I had hoped :P Oops.

Will fix.
